src/components/CoursePreview/index.js

```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import StyledCoursePreview from './styles';
+
+export default function CoursePreview({ course }){
+    return (
+      <StyledCoursePreview>
+        <Link to={`curso/${course.id}`}>
```

Comment:
Coloca uma `/` ao início do Link. Se por algum motivo esse componente for utilizado num outro que está em `/a/b/c` o usuário vai para `/a/b/curso/123`.